src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -6338,6 +6339,10 @@ private static void processNode(final TagNode node, final TagStatus status) {
 
       if (handleCosmicBowlingBall(str)) return;
 
+      if (FightRequest.handleTrainset(node)) {
```

Comment:
I think you could condition this on `status.toyTrain`.

Also the way this is written we have different places handling the logging and the preferences update -- could they be combined?

Reply:
Huh, I didn't notice that. Fixed.

Reply:
you marked this as resolved, but don't seem to have actually pushed the changes?

Reply:
Whoops